vulcanai2/tests/models/test_cnn.py

```diff
@@ -136,5 +139,19 @@ def test_fit_multi_input(self, multi_input_cnn,
                           range(len(multi_input_cnn_data)//2, 
                                     len(multi_input_cnn_data)))
         test_train_loader = DataLoader(test_train, batch_size=2)
-        test_val_loader = DataLoader(test_val, batch_size=2)
-        multi_input_cnn.fit(test_train_loader, test_val_loader, 1)
\ No newline at end of file
+        test_val_loader = DataLoader(test_val, batch_size=2)     
+        
+        init_weights = copy.deepcopy(multi_input_cnn.network[0]._kernel.weight.data)
+        multi_input_cnn_no_fit = copy.deepcopy(multi_input_cnn)
+        try:
+            multi_input_cnn.fit(test_train_loader, test_val_loader, 2)
+        except RuntimeError:
+            print("The network multi_input_cnn failed to train.")
+        finally:
+            trained_weights = multi_input_cnn.network[0]._kernel.weight.data.cpu()
+            
+            # Sanity check if the network parameters are training
+            assert (torch.equal(init_weights.cpu(), trained_weights.cpu()) is False)
```

Comment:
I don't think you need to specify cpu on this

Reply:
It throws error if I don't, since the model is in cuda in `test_forward_multi_input`

Reply:
have you ran all the tests on just CPU to confirm there's no issues?

Reply:
yeah but by default it has to be fed as a tuple. Take a look at here as well: https://discuss.pytorch.org/t/interpreting-gradcheck-errors/16239/11

Reply:
are you getting any errors? Not sure how do I make pytorch to ignore my gpu rather than uninstalling CUDA from my machine. Manually switch of device tests are already carried out in `test_device.py` I don't get any errors there.

Reply:
tests pass for just CPU can confirm